### PR TITLE
MM-T615 Only public channels available in outgoing web-hooks setup

### PR DESCRIPTION
**Pull request by MikeworX**
**_Summary_**

This PR validates that only public channels are displayed in the outgoing web-hook's channel select drop-down box. In addition, a new api command is added to cypress, **cy.apiRemoveUserFromChannel()**.

**_Steps_**

A regular user creates a new team as well as:
- a second user for a direct message channel
- a public channel they are a member of
- a public channel they are not a member of
- a private channel
- a public channel that is immediately archive
- it is then asserted that only the public channel they are a member of, appears in the drop-down.

**_NOTE_**

The archive test above will currently fail due to this [bug](https://mattermost.atlassian.net/browse/MM-30069). 

[vercel](https://automation-test-cases.vercel.app/test/MM-T615)
[TM4J](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T615)

### CODE REVIEW

**Review by josephbaylon — CHANGES_REQUESTED** [review]
@MikeworX I'm getting the following error below consistently. Also just minor comment suggestions.

Screen Shot 2020-10-30 at 11 00 04 AM

File: e2e/cypress/integration/integrations/outgoing_webhook/only_public_channels_available_in_webhook_spec.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        //* Assert that a public channel the user is not a member of does not appear in the dropdown
+        cy.get('#channelSelect').children().should('not.contain', nonMemberPublicChannel);
+
+        // * Assert that a private channel does not display in the drop-down
+        cy.get('#channelSelect').children().should('not.contain', privateChannel);
+
+        // * Assert that an archived / deleted channel does not appear in the drop-down // currently this fails due to a bug
+        cy.get('#channelSelect').children().should('not.contain', archiveChannel);
+
+        // Assert that a direct message channel does not appear in the drop-down
```
[the remark this answers is not in the document]
Comment (MikeworX) [reply e2e/cypress/integration/integrations/outgoing_webhook/only_public_channels_available_in_webhook_spec.js:80]: Thanks @josephbaylon for the review! The test is failing at that point due to this [bug](https://mattermost.atlassian.net/browse/MM-30069).

**Comment by jfrerich** [comment]
Hi @MikeworX! Is this ready for a re-review?

**Comment by MikeworX** [comment]
@jfrerich Hi there - this one will fail until https://mattermost.atlassian.net/browse/MM-30069 is fixed. Should I just close this PR and wait on the ticket?

**Comment by jfrerich** [comment]
@MikeworX Thanks for letting me know! I'll apply the appropriate label letting us know it is waiting on another PR. You don't need to close this PR :)

**Comment by lindy65** [comment]
Hi @MikeworX - I'm helping review stale PRs and will be closing this PR for now as https://mattermost.atlassian.net/browse/MM-30069 is still not in progress. I've sent the ticket back to Global Triage for prioritization.

//cc @saturninoabril @josephbaylon

**Pull request closed without merging**